src/main/java/aspects/LockAspect.java

```diff
@@ -96,12 +110,42 @@ private void logLockError(AuthenticatedRequestBean bean, ProceedingJoinPoint joi
     private Lock getLockAndBlock(String username) throws LockError {
         Lock lock = userLockRegistry.obtain(username);
         if (obtainLock(lock)) {
+            log.info("Obtained lock for key " + username);
             return lock;
         } else {
+            String holdingThreadName = getHoldingThread(username);
+            if (!threadLives(holdingThreadName)) {
+                log.error("Evicting thread " + holdingThreadName + " from lock with key " + username);
+                evictLock(username);
+                return getLockAndBlock(username);
+            }
             throw new LockError();
         }
     }
 
+    private boolean threadLives(String name) {
```

Comment:
And since I'm totally unfamiliar with this, how sure are you that this works? I.e. it'd be a mess if threads that were actually still running didn't show up in the list for some reason.